docs/arch/adr-25.md

```diff
@@ -0,0 +1,154 @@
+
+### Sub-task UI
+
+The sub-task UI positioned itself fixed below relative to the position of the mark. Notably transcription projects have commented that this interferes with being able to transcribe successfully since the dialog may cover up part of the subject and cannot be moved without moving the drawing mark.
+
+## Decision
+
+For initial support, we will support the single and multiple choice question tasks and the text task in the sub-task. Slider task may be deprecated and dropdown task may be changing in ways we do not have a plan for yet, so they can be supported later if it makes sense to add them.
+
+### Annotation JSON structure
+
+The annotations in the details array will be updated to be an object that just contains a reference to the task's unique identifier. The task annotation itself will be stored in the classification's annotations flattened. 
+
+The main benefit of this reorganization will be with downstream analysis and aggregation. When aggregating drawn shapes the first step is clustering. Once the clusters are found the subtasks need to be aggregated within each cluster. This will be easier to do if the structure of each subtask annotation is the same as if that task was asked on its own. The code can just take all subtask annotations within a cluster and just pass it to the reducer as if it is a list of main task annotations without having to reshape them.
+
+An example of the new sub-task annotation JSON structure:
+
+```json
+{"annotations": [
+  {
+    "task": "T0",
+    "taskType": "drawing",
+    "value": [
+      {
+        "frame": 0,
+        "tool": 0,
+        "toolType": "point",
+        "x": 452.18341064453125,
+        "y": 202.87478637695312,
+        "details": [
+          {"task": "T0.0.0.0"},
+          {"task": "T0.0.0.1"}
+        ]
+      },
+      {
+        "frame": 0,
+        "tool": 1,
+        "toolType": "point",
+        "x": 404.61279296875,
+        "y": 583.4398803710938,
+        "value_index": 1,
+        "details": [
+          {"task": "T0.1.1.0"},
+          {"task": "T0.1.1.1"}
+        ]
+      }
+    ]
+  },
+  {
+    "task": "T0.0.0.0",
+    "taskType": "single",
+    "value": 0
+  },
+  {
+    "task": "T0.0.0.1",
+    "taskType": "dropdown",
+    "value":[
+      {"option-1": 1},
+      {"option-2": 1},
+      {"None": 1}
+    ]
+  },
+  {
+    "task": "T0.1.1.0",
+    "taskType": "single",
+    "value": 1
+  },
+  {
+    "task": "T0.1.1.1",
+    "taskType": "dropdown",
+    "value": [
+      {"option-3": 1},
+      {"option-4": 1},
+      {"option-5": 1}
+    ]
+  }
+]}
+```
+
+The sub-task identifiers follow a convention of `TASK_KEY.ANNOTATION_INDEX.TOOL_INDEX.DETAILS_INDEX`.
```

Comment:
The naming is confusing here. There's only one annotation in the example, with two marks (I think.) Is `ANNOTATION_INDEX` the index of a particular mark inside the annotation? If so, I'd suggest `MARK_INDEX` instead.

Reply:
Maybe change the order to task > tool > mark > detail as well? That reflects the hierarchy in the drawing model.

Reply:
On reflection, this isn't a task identifier, since it points to a specific mark. I think Coleman's picked this up in his comment, but `annotation.task` should point to the task definition, so that analysis code knows how to parse `annotation.value`. In this case, the task definition is stored in Panoptes in `workflow.tasks[taskKey].tools[toolIndex].details`, rather than on each mark.

If all we need to analyse `annotation.value` is the task type, then this can be fixed by adding `annotation.type` to annotations. It depends whether any other info from the task is needed.

Reply:
I would be happy changing it to a true task identifier (e.g. `TASK_KEY.TOOL_INDEX.DETAILS_INDEX`) and having another field with the mark identifier `ANNOTATION_INDEX`.

Reply:
If we're going to use this as the model while drawing is in progress, then the mark identifier will need to be a value that doesn't change as we add and remove marks.

If this model is only used to save completed classifications to Panoptes, then annotation index will work just fine.

My vote goes with the latter option, because a consequence of the former is breaking changes to the classifications store model, even for workflows that don't use the drawing task.

Reply:
My vote goes for latter too because we can't write JS maps to Panoptes anyway, and you've noted that marks are maps in the MST models, so we will be doing some transformations at classification submission time regardless.

Reply:
Cool. In that case, I think #1376 can be reviewed independently of this ADR, because that PR doesn't touch on how classifications are generated from marks.

I think that will make working with subtasks cleaner too. We can annotate a subtask with `mark.addAnnotation(task, value)` while continuing to use `classification.addAnnotation(task, value)` to annotation classifications. (Noting that `task` in the classifier store API is an actual Task model, not a task key.)

Reply:
I'll add a note about the suggestions here being the annotation model at submission of the classification.

Reply:
Ok, I've made some changes that include clarifying that this is a proposal for the classification annotation JSON structure at submission and added `markIndex` to the sub-task annotation structure. I used `markIndex` instead of `annotationIndex` to be clear we're referring to an individual drawing task's `value` array rather than the parent `annotations` array. I sometimes conflate an individual's drawing task's `value` array which represents the marks with the parent `annotations` array in describing this. Is this clear or just more confusing?

Reply:
@CKrawczyk and @eatyourgreens is the most recent text satisfactory?

Reply:
I'll take no response as acceptance, but we have issue #1443 where this can be discussed further.